apps/query-ui/queryui/queryui.py

```diff
@@ -18,13 +18,22 @@
 
 
 DEFAULT_S3_CACHE_PATH = "s3://aryn-temp/llm_cache/luna/ntsb"
-
-
-def execute(code: str):
-    try:
-        exec(code, globals(), globals())
-    except Exception as e:
-        st.exception(e)
+DEFAULT_TRACE_DIR = "luna_traces"
+DEFAULT_FULL_TRACE_DIR = f"{os.path.dirname(os.getcwd())}/query-ui/luna_traces"
```

Comment:
I don't want the demo UI to scribble onto the local directory. We should always use tempdir.mkdtemp() for this, which is what the current code is doing - can't you retain the use of mkdtemp and set st.session_state.trace_dir to the directory name that is generated?